### ISSUE TITLE
ReduceLROnPlateau scheduler type check

### ISSUE THREAD

**Issue by alexeykarnachev** [PyTorchLightning/pytorch-lightning#1114]
## 🐛 Bug

Incorrect type check for scheduler of class ReduceLROnPlateau.
https://github.com/PyTorchLightning/pytorch-lightning/blob/bc01b9ac1c0ddf503f83755c00d5541a2679e5b4/pytorch_lightning/trainer/trainer.py#L713

I believe, this check:
`isinstance(scheduler, optim.lr_scheduler.ReduceLROnPlateau)`

must look like this:
`isinstance(scheduler['scheduler'], optim.lr_scheduler.ReduceLROnPlateau)`

### To Reproduce

1. Create a scheduler of type `optim.lr_scheduler.ReduceLROnPlateau` in the `configure_optimizers` method of a `LightningModule` class.
2. Return an optimizer and scheduler from this method. Place them in lists: `return [optimizer], [scheduler]`.
2. Execute the `trainer.fit(module)`.
3. Put a break-point here: https://github.com/PyTorchLightning/pytorch-lightning/blob/bc01b9ac1c0ddf503f83755c00d5541a2679e5b4/pytorch_lightning/trainer/trainer.py#L712
4. Make sure that the condition is never True.

**Comment by Borda** [comment]
@SkafteNicki could you pls chek it?

**Comment by SkafteNicki** [comment]
I agree with @alexeykarnachev that this is indeed a bug, that I introduced with the recent rework of the learning rate schedulers. I can send a new PR with the bug fix and include a test for `ReduceLROnPlateau` schedulers

**Comment by Borda** [comment]
Thx guys!
@alexeykarnachev minds send fix PR?
